java/ql/src/semmle/code/java/dataflow/internal/DataFlowUtil.qll

```diff
@@ -368,6 +368,60 @@ predicate hasNonlocalValue(FieldRead fr) {
  */
 predicate localFlowStep(Node node1, Node node2) { simpleLocalFlowStep(node1, node2) }
 
+/**
+ * A method or constructor that returns the exact value of one of its parameters or the qualifier.
+ *
+ * Extend this class and override `returnsValue` to add additional value-preserving steps through a
+ * method that should be added to the basic local flow step relation.
+ *
+ * These steps will be visible for all global data-flow purposes, as well as via
+ * `DataFlow::Node.getASuccessor` and other related functions exposing intraprocedural dataflow.
+ */
+abstract class ValuePreservingCallable extends Callable {
```

Comment:
Using an abstract class for this comes with a heavy commitment to bidirectional import. Therefore, I don't think this is the right place for this. Instead we should put both of these abstract classes in `FlowSteps.qll`.
The few that are already modelled here (`StandardLibraryValuePreservingCallable`) should move to a different place as well: Either directly in `FlowSteps.qll` as that already contains a bit of JDK modelling, or quite possibly moved to a new file along with the other modelling from `FlowSteps.qll`.

Reply:
Moved them both to FlowSteps.qll